### PR TITLE
Docs/state management api

### PR DESCRIPTION
**Pull request by mtomcal**
- Split out API docs into multiple Markdown files as I ran out of Hierarchy for headers
- Added State Management API docs

- [x] Update links based on future @10xLaCroixDrinker Holocron PR
- [x] Update links based on future @10xLaCroixDrinker One App Ducks PR

### CODE REVIEW

File: docs/api/modules/State-Management.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+[👈 Return to Overview](../API.md)
+
+# State Management
```
Comment (mtomcal) [inline docs/api/modules/State-Management.md]: This is where the new content starts for State Management API.

File: docs/api/modules/State-Management.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+[Holocron Modules](../API.md#modules) rely on one Store provided by [Redux](https://redux.js.org/) used primarily to cache the results of loaded data and Modules. Every Module may add their own [Reducers](https://redux.js.org/basics/reducers/) to the shared Store (through [`holocronModule`](https://github.com/americanexpress/holocron/blob/master/packages/holocron/API.md#holocronmodule)) and may dispatch actions to transform the Store. Redux provides a simple way to store data on the Server and [preload the store](https://redux.js.org/recipes/server-rendering#inject-initial-component-html-and-state) used by the Browser. This is commonly referred to as [Server Side Rendering](https://redux.js.org/recipes/server-rendering). One App employs the [Ducks Specification](https://github.com/erikras/ducks-modular-redux) design pattern for the logical grouping of Reducers, Action Creators, and Selectors (e.g. the `error` ducks contains reducers, actions, and selectors for storing error data in the Redux Store).
```
Comment (10xLaCroixDrinker) [inline docs/api/modules/State-Management.md]: "`error` ducks" should be "`error` duck"

File: docs/api/modules/State-Management.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+* ✅ Server
+* ✅ Browser
+
+**Shape**
+
+```js
+global.BROWSER; // Boolean
+```
+
+The `global.BROWSER` is provided in the Server and Browser environments to determine if the code is currently being executed on the Server or the Browser.
```
Comment (10xLaCroixDrinker) [inline docs/api/modules/State-Management.md]: get rid of leading "The"

File: docs/api/modules/State-Management.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+## Higher Order Components
+
+[Holocron Modules](../API.md#modules) use [Higher Order Components (HOC)](https://reactjs.org/docs/higher-order-components.html) to add behaviors regarding 1) when a Module loads, 2) connecting a Module with its [Reducer(s)](https://redux.js.org/basics/reducers/) to a Redux Store (similar to [Redux `connect`](https://react-redux.js.org/api/connect)) and 3) adding runtime validations for a Module.
```
Comment (10xLaCroixDrinker) [inline docs/api/modules/State-Management.md]: I'm not a fan of the inline numbering. Remove the numbers or make it an ol

File: docs/api/modules/State-Management.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    // URL where the Browser sends client side errors to.
+    reportingUrl: String,
+    // URL where the One App static assets are located.
+    cdnUrl: String,
+    // The exact filename for the locale file used in the Browser.
+    localeFilename: String,
+    // Name of the Holocron Module that serves as the entry point to your application.
+    rootModuleName: String,
+    // ... Settings from provideStateConfig key values will land here.
+    [provideStateConfigSettingName]: String,
```
Comment (10xLaCroixDrinker) [inline docs/api/modules/State-Management.md]: Don't we parse these values, so they are not always strings?

File: docs/api/modules/State-Management.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+  * [`ONE_CLIENT_LOCALE_FILENAME`](#),
+  * [`ONE_CLIENT_ROOT_MODULE_NAME`](#)
+
+#### Action Creators
+
+##### `setConfig`
+
+> ⚠️ For Internal Use
+
+**Shape**
```
Comment (10xLaCroixDrinker) [inline docs/api/modules/State-Management.md]: shape? I'd expect this to be `**Usage**`, followed by `**Arguments**` after the code block

File: docs/api/modules/State-Management.md (hunk 5 above)
Comment (mtomcal) [reply docs/api/modules/State-Management.md]: They are always strings. The values that land in `config` are set based on environmental variable and browser/server. 

Parsing `stateConfig.js`:
https://github.com/americanexpress/one-app/blob/master/src/server/utils/stateConfig.js#L127-L140

Calling `setConfig` action:
https://github.com/americanexpress/one-app/blob/0c64dcd7b76018868c37101e50ef2228c25a3eec/src/server/middleware/sendHtml.js#L287

File: docs/api/modules/State-Management.md (hunk 6 above)
Comment (mtomcal) [reply docs/api/modules/State-Management.md]: We use `**Shape**` elsewhere: https://github.com/americanexpress/one-app/pull/19/files/1f14dc8aabda592857072b3822bee43cb70de4b9#diff-2919092b5b3a7dfbd26a8e11b7650597R17-R20

Ill add `**Arguments**` header.

File: docs/api/modules/State-Management.md (hunk 4 above)
Comment (mtomcal) [reply docs/api/modules/State-Management.md]: I really like inline numbering and I've used a few times elsewhere:
* https://github.com/americanexpress/one-app/pull/19/files/1f14dc8aabda592857072b3822bee43cb70de4b9#diff-16c8bdba86123da609a7dc1cdf2fccbbR321
* https://github.com/americanexpress/one-app/pull/19/files/1f14dc8aabda592857072b3822bee43cb70de4b9#diff-2919092b5b3a7dfbd26a8e11b7650597R29
* https://github.com/americanexpress/one-app/pull/19/files/1f14dc8aabda592857072b3822bee43cb70de4b9#diff-affc201ede6b1995b07d8f7a823a5dcdR5

It allows paragraph form to be more easily parsable to the eye and helpful in technical writing IMO. If you insist, Ill change it all.

File: docs/api/modules/State-Management.md (hunk 5 above)
Comment (10xLaCroixDrinker) [reply docs/api/modules/State-Management.md]: I was thinking of the env var parsing https://github.com/americanexpress/one-app/blob/master/src/server/utils/stateConfig.js#L23-L37

We should definitely make this more flexible

Comment (10xLaCroixDrinker) [reply docs/api/modules/State-Management.md]: The env vars are parsed, so they are not always strings

https://github.com/americanexpress/one-app/blob/master/src/server/utils/stateConfig.js#L23-L37

The config from the root module should be more flexible

File: docs/api/modules/State-Management.md (hunk 4 above)
Comment (10xLaCroixDrinker) [reply docs/api/modules/State-Management.md]: I find it less readable. Maybe a third opinion is needed

Comment (JamesSingleton) [reply docs/api/modules/State-Management.md]: I find it less readable as well. In my mind as I read it, the numbers end up blending in with everything else and if they are there to make a point, it doesn't get the point across to me.

Comment (mtomcal) [reply docs/api/modules/State-Management.md]: Democracy has spoken, I will remove the inline numbers.

**Pull request merged**